### PR TITLE
Added support for mysql 

### PR DESCRIPTION
**Pull request by amonetta**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
Added support to MySQL/MariaDB in plug-in SQL.

**Description of change**
Added initializer and example configuration for MySQL connection.
Rewrites model to support `REFERENCES` in standard SQL language

**Which issue this PR fixes**
Just an improvement.

### CODE REVIEW

**Comment by evan2645** [comment]
hey @amonetta - big thanks for this contribution!

I noticed that you have a few PRs open - should I close the other two?

Also, it looks like maybe you're still working on this? Are you ready for us to review it or we should wait?

File: doc/plugin_server_datastore_sql.md [hunk 1]
```diff
@@ -57,3 +57,27 @@ connection_string="dbname=postgres user=postgres password=password host=localhos
   the server was signed by a trusted CA and the server host name
   matches the one in the certificate)
 
+  ### `database_type = "mysql"`
+
+  This configuration is valid for MySQL and mariadb.
+  The `connection_string` for the PostreSQL database connection is kind of DB URL as
```
Comment (martincapello) [inline doc/plugin_server_datastore_sql.md]: Shouldn't it say "for the MySQL database..."?

**Comment by mweissbacher** [comment]
This looks pretty good, any updates on merging this?

**Comment by mcpherrinm** [comment]
This PR is obsoleted by https://github.com/spiffe/spire/pull/784

**Pull request closed without merging**